gradio/themes/base.py

```diff
@@ -32,6 +42,146 @@ def repl_func(match):
         dark_css += "}"
         return css + "\n" + dark_css
 
+    def to_dict(
+        self,
+    ):
+        """Hacky implementation for now."""
+        schema = {"theme": {}}
+        for prop in dir(self):
+            if not prop.startswith("_") and isinstance(getattr(self, prop), str):
+                schema["theme"][prop] = getattr(self, prop)
+        return schema
+
+    @classmethod
+    def load(cls, path: str) -> "ThemeClass":
+        theme = json.load(open(path))
+        return cls.from_dict(theme)
+
+    @classmethod
+    def from_dict(cls, theme) -> "ThemeClass":
+        base = cls()
+        for prop, value in theme["theme"].items():
+            setattr(base, prop, value)
+        return base
+
+    def dump(self, path: str, filename: str):
+        as_dict = self.to_dict()
+        json.dump(as_dict, open(Path(path) / Path(filename), "w"))
+
+    @classmethod
+    def from_hub(cls, repo_name: str):
+        if "@" not in repo_name:
+            name, version = repo_name, None
+        else:
+            name, version = repo_name.split("@")
+
+        assets = get_theme_assets(name)
+        matching_version = get_matching_version(assets, version)
+
+        if not matching_version:
+            raise ValueError(
+                f"Cannot find a matching version for expression {version} "
+                f"from files {[f.filename for f in assets]}"
+            )
+
+        url = huggingface_hub.hf_hub_url(
+            repo_id=name,
+            repo_type="space",
+            filename=f"themes/theme_schema@{matching_version.version}.json",
+        )
+        download = requests.get(url)
+        if download.ok:
+            return cls.from_dict(download.json())
+        else:
+            raise ValueError(
+                f"Error downloading vile themes/theme_schema@{matching_version}.json from {repo_name}!"
+            )
+
+    def to_hub(
+        self,
+        repo_name: str,
+        version: str,
+        hf_token: str,
+        theme_name: str | None = None,
+        description: str | None = None,
+    ):
+
+        # To check if version is valid string
+        _ = semver.Version(version)
+
+        from gradio import __version__
+        from gradio.themes import app
+
+        api = huggingface_hub.HfApi()
+
+        author = api.whoami(hf_token)["name"]
+
+        huggingface_hub.create_repo(
```

Comment:
This is the structure of the uploaded space

https://huggingface.co/spaces/freddyaboulton/dracula_revamped/tree/main

<img width="1403" alt="image" src="https://user-images.githubusercontent.com/41651716/224199329-a0c81363-d2f3-4156-a245-143a3cb7032e.png">

All of the theme versions are in the `themes` directory.

<img width="746" alt="image" src="https://user-images.githubusercontent.com/41651716/224199389-d60d3d83-9414-4ade-bd0d-7771fce63588.png">


We had talked about using different branches for the different versions but I think it will be clearer and more discoverable if everything is on the main branch